Plots/Shapefiles/NCL_shapefiles_1.py

```diff
@@ -0,0 +1,109 @@
+open(gdf.get("shape_files/states.dbf"), 'r')
+open(gdf.get("shape_files/states.shp"), 'r')
+open(gdf.get("shape_files/states.shx"), 'r')
+open(gdf.get("shape_files/states.prj"), 'r')
+
+# Open shapefiles
+shapefile = shp.Reader(gdf.get("shape_files/states.dbf"))
+
+###############################################################################
+# Set color map colors and bounds
+colormap = colors.ListedColormap(['blue', 'lime', 'yellow', 'red'])
+
+colorbounds = [0.5, 1.5, 2.5, 3.5, 4.5]
+
+norm = colors.BoundaryNorm(colorbounds, colormap.N)
+###############################################################################
+# Helper function to determine state color:
+def color_assignment(record):
+    population = record.PERSONS
+    unempolyment = record.UNEMPLOY
+    percent = unempolyment / population
+    if (0.01 <= percent and percent < 0.02):
+        return colormap.colors[0]
+    elif (0.02 <= percent and percent < 0.03):
+        return colormap.colors[1]
+    elif (0.03 <= percent and percent < 0.04):
+        return colormap.colors[2]
+    elif (0.04 <= percent):
+        return colormap.colors[3]
+
+###############################################################################
+# Plot:
+plt.figure(figsize=(10,8))
+ax = plt.axes(projection=ccrs.LambertConformal(standard_parallels=(33, 45), central_longitude=-98))
+ax.set_extent([-125, -74, 22, 50])
+
+ax.add_feature(cfeature.LAND, color='silver', zorder=0)
+ax.add_feature(cfeature.LAKES, color='white', zorder=1)
+
+for i in range(0, len(shapefile.shapes())):
+    shape = shapefile.shape(i)
+    record = shapefile.record(i)
+    color = color_assignment(record)
+    if len(shape.parts) > 1: # if a shape has multiple parts make each one a separate patch
+        for j in range(0, len(shape.parts)):
+            start_index = shape.parts[j]
+            if (j is (len(shape.parts)-1)): # the last part uses the remaining points and doesn't require and end_index  
+                patch = mpatches.Polygon(shape.points[start_index:], facecolor=color, edgecolor='black', linewidth=0.5, transform=ccrs.PlateCarree(), zorder=2)
+            else:
+                end_index = shape.parts[j+1]
+                patch = mpatches.Polygon(shape.points[start_index : end_index], facecolor=color, edgecolor='black', linewidth=0.5, transform=ccrs.PlateCarree(), zorder=2)
+            ax.add_patch(patch)
+    else:
+        patch = mpatches.Polygon(shape.points, facecolor=color, edgecolor='black', linewidth=0.5, transform=ccrs.PlateCarree(), zorder=2)
+        ax.add_patch(patch)
+
+# Create colorbar
+plt.colorbar(cm.ScalarMappable(cmap=colormap, norm=norm), ax=ax, boundaries=colorbounds, orientation='horizontal', shrink=0.75, ticks=[1, 2, 3, 4], label='percent', aspect=30, pad=0.075)
+
+# Add latitude and longitude labels
+gl = ax.gridlines(draw_labels=True, x_inline=False, y_inline=False)
+gl.xlocator = mticker.FixedLocator(np.linspace(-120, -80, 5))
```

Comment:
Is there any way you can set the top/bottom x-axes, and the left/right y-axes to have different tick mark locations? For example, having the bottom x-axis range from -120 to -80 but the top x-axis range from -135 to -75, like in the NCL plot.

Reply:
I'm not sure, but I'll see if I can figure it out.

Reply:
@clairefio Different labels for the top/bottom and left/right sides of maps doesn't seem to be supported in cartopy. I could turn off the labels being drawn on the top and right side and then add the labels in manually but I'm not sure if it's worth it since the plot is legible as is. What do you think?

Reply:
I've run into that problem as well. I think how you have it now should be fine!